### ISSUE TITLE
different shape/size = black

### ISSUE THREAD

**Issue by samim23** [stitchfix/fauxtograph#3]
I´m testing out the --shape param, attempting to train/generate higher resolution images.
Seems like it does not like square formats, unfortunately resulting in black images. 
Getting a bunch of these errors to: "RuntimeWarning: overflow encountered in exp self.y = 1 / (1 + numpy.exp(-x[0]))"

Any tips here?

**Comment by tjtorres** [comment]
The way the CLI is structured, it's a bit hard to change just one parameter rapidly(especially moving to much higher resolutions). Many times going to higher resolutions leads to greater errors and the batch size needs to be reduced (or the learning rate reduced) at the beginning of the training. Effectively, when the random weights are generated it makes for KL Divergence terms which can be quite large to start and that can lead to convergence on minima which are non-optimal as well as error terms which overflow the max float32 size (which is probably why you're getting the RuntimeWarning). If you want to go to higher resolutions I'd recommend playing around with the parameters at the module level rather than using the command line tool. I mostly included that as a fun toy example so that people can get started quickly, but the customizability there is limited. 

    For higher resolutions, the right solution (and one I will likely end up implementing eventually) is to start with convolution layers in order to reduce the size of the final layer before moving to a fully connected system. Otherwise the memory allocation and training times needed, in order to avoid stepping down in layer size too rapidly after the input layer, can be prohibitive.

   Try customizing the layer sizes, batch size, loss ratio between reconstruction loss and KL terms, and latent width with the module and see if that works. Additionally, it is a probabilistic method which starts off with a random set of layer weights, so just retrying the training step a few times may possibly net you a better model. Sorry if that isn't more satisfying.

**Comment by samim23** [comment]
Been tuning the hyper-parameters a bit (of course not using CLI) but as you pointed out, its tricky to get right. Using a convolutional auto-encoder (something like https://github.com/mikesj-public/convolutional_autoencoder) does indeed sound promising, or even using a Generative Adversarial Network. Exploring VAE in-depth is a good first step though to build intuition.
